### ISSUE TITLE
Make systemd dependency optional

### ISSUE THREAD

**Issue by smortex** [Snawoot/postfix-mta-sts-resolver#52]
**Is your feature request related to a problem? Please describe.**
systemd is only available on GNU/Linux systems, but users of other kernels / operating systems (e.g. FreeBSD) might still get benefits from this software.

**Describe the solution you'd like**
The sdnotify dependency added in #16 is unconditional: if sdnotify cannot be used, the daemon will not run.  Making this an optional dependency would allow to run the software on FreeBSD unpatched.

Thanks !

**Comment by Snawoot** [comment]
Hi!

sd_notify is a very simple generic text protocol, which allows status signaling from daemon if it sees appropriate socket in it environment. Practically, you can benefit from it even on FreeBSD, if your supervising daemon or shell wrapper supports it.

pmsr already runs just fine on platforms which do not have systemd without any patches. For example, it's Docker (Alpine Linux), Windows, MacOS and *BSD. The only problem is it requires presence of very small sdnotify Python package in order to detect if sdnotify required and send status update. It is some sort of sdnotify feature: you don't need separate configuration to enable it, it is enabled automatically when it is appropriate.

If you are just using this package installed from pip, you will not notice any issues with sdnotify.

If you are distro package/port maintainer, you have two options:
1. Just provide Python sdnotify package. Why not? It's usable even on FreeBSD.
2. Maintain distro-specific patches. It's a pretty common practice.

Personnaly, I think first option is versatile and acceptable for all cases.

**Comment by Snawoot** [comment]
I think I've found solution which will be convenient for everyone. I'll reimplement sd_notify support module. It will result native asyncio support for this module and this module will be bundled in pmsr package, so there will be no dependency for external sdnotify package. This way it'll just work everywhere where suitable.

**Comment by smortex** [comment]
Ah, I was precisely starting to create a port for it when I realized it was systemd releated and probably not making any sense on FreeBSD (we just run the Python script with `daemon(1)`, which basically detach the process and allows us to check if the process is still running or not).

For now, I'll stick with the patches solution (because it's something I have already committed to the FreeBSD ports tree).  If the dependency can be trimmed in some way, I will just remove these patches.  If it happen it's not that simple, I will consider looking into sdnotify deeper… Maybe it make sense to add support for these notification messages to our `dameon(1)` implementation :wink:

Thank you for the feedback!
